### PR TITLE
Update Troubleshooting docs

### LINKED ISSUE #226: Basic debugging instructions for tutorial/any rad app
We need to help users debug their applications and environment in case of issues. 
There are various Radius system involved in deploying an application - would be good to tell users "What logs to check if you are encountering x type of issue)

### PR DESCRIPTION
**Pull request by mishrapratikshya**
Fixes https://github.com/project-radius/docs/issues/226

### CODE REVIEW

File: docs/content/reference/troubleshooting-radius.md [hunk 1]
```diff
@@ -45,3 +45,50 @@ try some of these solutions:
 
 Issues faced when deploying applications
 
+## Troubleshooting Radius Service 
+
+To troubleshoot radius serice user has to fetch the logs from the radius service containers
```
Comment (willdavsmith) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
To troubleshoot the Radius service, the user should fetch the logs from the Radius service containers.
```

File: docs/content/reference/troubleshooting-radius.md [hunk 2]
```diff
@@ -45,3 +45,50 @@ try some of these solutions:
...
+
+1. Run the kubectl command to list the applications running in your cluster under the namespace radius-system to get the pod details of appcore-rp, ucp and deployment engine.
+
+```
+kubectl get pods -n radius-system
+```
+
+2. Get the logs from radius appcore-rp, ucp and deployment engine container by using the pod names captured from the previous step and running the following command.
```
Comment (willdavsmith) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
2. Get the logs from radius `appcore-rp`, `ucp` and `bicep-de` containers by using the pod names captured from the previous step and running the following command.
```

File: docs/content/reference/troubleshooting-radius.md [hunk 3]
```diff
@@ -45,3 +45,50 @@ try some of these solutions:
 
 Issues faced when deploying applications
 
+## Troubleshooting Radius Service 
+
+To troubleshoot radius serice user has to fetch the logs from the radius service containers
+
+1. Run the kubectl command to list the applications running in your cluster under the namespace radius-system to get the pod details of appcore-rp, ucp and deployment engine.
```
Comment (willdavsmith) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
1. Run the `kubectl` command to list the pods running in your cluster under the namespace `radius-system` to  see that `appcore-rp`, `ucp`, and `bicep-de` pods are running.
```

File: docs/content/reference/troubleshooting-radius.md [hunk 4]
```diff
@@ -45,3 +45,50 @@ try some of these solutions:
...
+
+```
+kubectl get pods -n radius-system
+```
+
+2. Get the logs from radius appcore-rp, ucp and deployment engine container by using the pod names captured from the previous step and running the following command.
+
+```
+kubectl logs -f <appcore-rp pod name> -n radius-system
+kubectl logs -f <deployment engine pod name> -n radius-system
```
Comment (willdavsmith) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
kubectl logs -f <bicep-de pod name> -n radius-system
```

File: docs/content/reference/troubleshooting-radius.md [hunk 5]
```diff
@@ -45,3 +45,50 @@ try some of these solutions:
...
+kubectl logs -f <ucp pod> -n radius-system
```
Comment (willdavsmith) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
kubectl logs -f <ucp pod name> -n radius-system
```

File: docs/content/reference/troubleshooting-radius.md [hunk 6]
```diff
@@ -45,3 +45,50 @@ try some of these solutions:
...
+```
+
```
Comment (willdavsmith) [inline docs/content/reference/troubleshooting-radius.md:61]: You can also get the logs without the pod names like so:
```
k logs -n radius-system -l control-plane=de
k logs -n radius-system -l control-plane=ucp
k logs -n radius-system -l control-plane=appcore-rp
```

File: docs/content/reference/troubleshooting-radius.md [hunk 7]
```diff
@@ -45,3 +45,50 @@ try some of these solutions:
...
+3. Check for the text "panic" in the logs and if you find it open an issue at https://github.com/project-radius/radius by providing the details.
+
+## Troubleshooting issues with Azure Cloud Provider
+
+To troubleshoot issues with the Azure cloud provider and deployments to user's Azure subscription and resource group, refer to ARM logs. Log in to https://portal.azure.com and within the target resource group navigate to the "Activity logs" from the menu. For details refer https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/activity-log?tabs=powershell#view-the-activity-log  
+
+## Examples
+
+Deploying an application to Azure with a service principle with incorrect scope returns an error from deployment engine
```
Comment (willdavsmith) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
Deploying an application to Azure with a service principal with incorrect scope returns an error from deployment engine
```

**Review by willdavsmith — DISMISSED** [review]

File: docs/content/reference/troubleshooting-radius.md [hunk 8]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
 
 Issues faced when deploying applications
 
+## Troubleshooting Radius Service
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
## Troubleshooting the Radius control-plane
```

File: docs/content/reference/troubleshooting-radius.md [hunk 9]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
 
 Issues faced when deploying applications
 
+## Troubleshooting Radius Service 
+
+To troubleshoot the Radius service, the user should fetch the logs from the Radius service containers.
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
To troubleshoot the Radius control-plane, begin by viewing the logs from the [control-plane services]({{< ref architecture >}}) within the `radius-system` namespace. The kubectl CLI can be used, or a graphical tool such as [Octant](https://octant.dev/).
```

File: docs/content/reference/troubleshooting-radius.md [hunk 10]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+
+1. Run the `kubectl` command to list the pods running in your cluster under the namespace `radius-system` to  see that `appcore-rp`, `ucp`, and `bicep-de` pods are running.
+
+```
+kubectl get pods -n radius-system
+```
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]: Indent these 3 lines by 3 spaces in order to nest this code block under the numbered list (can't make a code suggestion due to the code block

File: docs/content/reference/troubleshooting-radius.md [hunk 11]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+
+2. Get the logs from radius `appcore-rp`, `ucp` and `bicep-de` containers by using the following command.
+
+```
+kubectl logs -n radius-system -l control-plane=de
+kubectl logs -n radius-system -l control-plane=ucp
+kubectl logs -n radius-system -l control-plane=appcore-rp
+```
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]: Indent these 5 lines by 3 spaces

File: docs/content/reference/troubleshooting-radius.md [hunk 12]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
 
 Issues faced when deploying applications
 
+## Troubleshooting Radius Service 
+
+To troubleshoot the Radius service, the user should fetch the logs from the Radius service containers.
+
+1. Run the `kubectl` command to list the pods running in your cluster under the namespace `radius-system` to  see that `appcore-rp`, `ucp`, and `bicep-de` pods are running.
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
1. Use the `kubectl` CLI to list the control-plane pods running in your cluster under the namespace `radius-system`. You should see the `appcore-rp`, `ucp`, and `bicep-de` pods running:
```

File: docs/content/reference/troubleshooting-radius.md [hunk 13]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+
+```
+kubectl get pods -n radius-system
+```
+
+2. Get the logs from radius `appcore-rp`, `ucp` and `bicep-de` containers by using the following command.
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
2. Get the logs from the `appcore-rp`, `ucp` and `bicep-de` containers by using the following commands:
```

File: docs/content/reference/troubleshooting-radius.md [hunk 14]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+
+```
+kubectl logs -n radius-system -l control-plane=de
+kubectl logs -n radius-system -l control-plane=ucp
+kubectl logs -n radius-system -l control-plane=appcore-rp
+```
+
+You can also get the logs by using the pod names captured from the previous step and running the following command.
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
   You can also get the logs by using the pod names captured from the previous step and running the following commands:
```

File: docs/content/reference/troubleshooting-radius.md [hunk 15]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+kubectl logs -n radius-system -l control-plane=appcore-rp
+```
+
+You can also get the logs by using the pod names captured from the previous step and running the following command. 
+
+```
+kubectl logs -f <appcore-rp pod name> -n radius-system
+kubectl logs -f <bicep-de pod name> -n radius-system
+kubectl logs -f <ucp pod name> -n radius-system
+```
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]: Indent this code block

File: docs/content/reference/troubleshooting-radius.md [hunk 16]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+
+3. Check for the text "panic" in the logs and if you find it open an issue at https://github.com/project-radius/radius by providing the details.
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
3. Look for the text "panic" or "error" in the logs, and if you find it inspect the error message. Also please open an issue at [project-radius/radius](https://github.com/project-radius/radius/issues/new?assignees=&labels=kind%2Fbug&template=bug.md&title=%3CBUG+TITLE%3E) with the details of your error and if possible, how to recreate.
```

File: docs/content/reference/troubleshooting-radius.md [hunk 17]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+
+## Troubleshooting issues with Azure Cloud Provider
+
+To troubleshoot issues with the Azure cloud provider and deployments to user's Azure subscription and resource group, refer to ARM logs. Log in to https://portal.azure.com and within the target resource group navigate to the "Activity logs" from the menu. For details refer https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/activity-log?tabs=powershell#view-the-activity-log
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
To troubleshoot issues with the [Azure cloud provider]({{< ref providers >}}) and deployments to a Microsoft Azure subscription and resource group, refer to your [ARM activity logs](https://docs.microsoft.com/azure/azure-monitor/essentials/activity-log):

1. Visit https://portal.azure.com
1. Navigate to your target subscription and resource group
1. Select "Activity logs" from the menu
1. Inspect the logs for any errors of failed deployments

Note that Activity logs may take up to 10 minutes to be available in Azure.
```

File: docs/content/reference/troubleshooting-radius.md [hunk 18]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+kubectl logs -f <ucp pod name> -n radius-system
+```
+
+3. Check for the text "panic" in the logs and if you find it open an issue at https://github.com/project-radius/radius by providing the details.
+
+## Troubleshooting issues with Azure Cloud Provider
+
+To troubleshoot issues with the Azure cloud provider and deployments to user's Azure subscription and resource group, refer to ARM logs. Log in to https://portal.azure.com and within the target resource group navigate to the "Activity logs" from the menu. For details refer https://docs.microsoft.com/en-us/azure/azure-monitor/essentials/activity-log?tabs=powershell#view-the-activity-log  
+
+## Examples
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
## Example: Troubleshooting a failed deployment
```

File: docs/content/reference/troubleshooting-radius.md [hunk 19]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+
+Deploying an application to Azure with a service principal with incorrect scope returns an error from deployment engine
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
In this example, a failure is returned after attempting to deploy a Bicep template that contains a Radius application plus Azure resources:
```

File: docs/content/reference/troubleshooting-radius.md [hunk 20]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+```
+rad deploy test.bicep
+Building test.bicep...
+Deploying Application into workspace 'w2'...
+Deployment In Progress...
+
+Error: ResourceDeploymentClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="InternalServerError" Message="Internal server error."
+```
+
+Deployment engine logs
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
This error isn't very descriptive. Let's take a look at the control-plane logs to see if they can tell us more:

```

File: docs/content/reference/troubleshooting-radius.md [hunk 21]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+```
+kubectl logs -f bicep-de-6d4546df86-rhthn -n radius-system
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
kubectl logs -n radius-system -l control-plane=de
```

File: docs/content/reference/troubleshooting-radius.md [hunk 22]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+
+fail: Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware[1]
+An unhandled exception has occurred while executing the request.
+Azure.RequestFailedException: The client '81082ee4-ace0-4922-a80a-2c02cb0d2207' with object id '81082ee4-ace0-4922-a80a-2c02cb0d2207' does not have authorization to perform action 'Microsoft.Resources/subscriptions/resourcegroups/read' over scope '/subscriptions/66d1209e-1382-45d3-99bb-650e6bf63fc0/resourcegroups/radius-rg-MZ94f' or the scope is invalid. If access was recently granted, please refresh your credentials.
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]:
```suggestion
Azure.RequestFailedException: The client 'APP_ID' with object id 'OBJECT_ID' does not have authorization to perform action 'Microsoft.Resources/subscriptions/resourcegroups/read' over scope '/subscriptions/SUBSCRIPTION_ID/resourcegroups/radius-rg-MZ94f' or the scope is invalid. If access was recently granted, please refresh your credentials.
```

**Review by emily-potyraj — DISMISSED** [review]
Looking good, thanks! 

I'm approving, but have have 1 non-blocking question, which is whether we should add a note about "In the future, the Radius CLI and Dashboard will have more native troubleshooting capabilities. For now, we still rely on kubernetes capabilities to help investigate issues."

File: docs/content/reference/troubleshooting-radius.md [hunk 23]
```diff
@@ -45,3 +45,58 @@ try some of these solutions:
...
+Status: 403 (Forbidden)
+ErrorCode: AuthorizationFailed
+```
```
Comment (AaronCrawfis) [inline docs/content/reference/troubleshooting-radius.md]: Add the following section please:

```
This tells us the service principal that was configured with the Azure cloud provider was not properly configured on the target resource group. You can now add your service principal to your resource group and try your deployment again.

Also please make sure to [open an Issue](https://github.com/project-radius/radius/issues/new?assignees=&labels=kind%2Fbug&template=bug.md&title=%3CBUG+TITLE%3E) if you encounter a generic `Internal server error` message, so we can address the root error not being forwarded to the user.
```

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]
Looking good! Couple formatting/wording suggestions but overall great content!

**Comment by mishrapratikshya** [comment]
> Looking good, thanks!
> 
> I'm approving, but have have 1 non-blocking question, which is whether we should add a note about "In the future, the Radius CLI and Dashboard will have more native troubleshooting capabilities. For now, we still rely on kubernetes capabilities to help investigate issues."

I think we should add it. @AaronCrawfis your thoughts?

**Review by willdavsmith — DISMISSED** [review]

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**